corehq/apps/locations/models.py

```diff
@@ -254,6 +311,12 @@ def _sync_location(self):
             }
         )
 
+        if is_new or (sql_location.location_type.name != self.location_type):
+            sql_location.location_type, _ = LocationType.objects.get_or_create(
+                domain=self.domain,
+                name=self.location_type,
+            )
```

Comment:
when do we anticipate that we will be creating a location without a valid type? i don't quite see how this should be possible, and it seems like this could just allow us to introduce other bugs in the code.